File: 1261-find-elements-in-a-contaminated-binary-tree/1261-find-elements-in-a-contaminated-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class FindElements:

    def __init__(self, root: TreeNode):

        def bfs(root: TreeNode) -> None:
            dq = collections.deque([root])
            self.seen.add(root.val)
            while dq:
                node = dq.popleft()
                if node.left:
                    node.left.val = 2 * node.val + 1
                    dq.append(node.left)
                    self.seen.add(node.left.val)
                if node.right:
                    node.right.val = 2 * node.val + 2
                    dq.append(node.right)
                    self.seen.add(node.right.val)
            
        self.seen = set()
        if root:
            root.val = 0
            bfs(root)
        
    def find(self, target: int) -> bool:
        return target in self.seen 

# Your FindElements object will be instantiated and called as such:
# obj = FindElements(root)
# param_1 = obj.find(target)

File: 1261-find-elements-in-a-contaminated-binary-tree/1261-find-elements-in-a-contaminated-binary-tree.js

var FindElements = function(root) {
    this.recoveredValues = new Set();
    root.val = 0;
    this.recoverTree(root);
};

FindElements.prototype.recoverTree = function(root) {
    if (!root) return;
    this.recoveredValues.add(root.val);
    if (root.left) {
        root.left.val = 2 * root.val + 1;
        this.recoverTree(root.left);
    }
    if (root.right) {
        root.right.val = 2 * root.val + 2;
        this.recoverTree(root.right);
    }
};

FindElements.prototype.find = function(target) {
    return this.recoveredValues.has(target);
};